lib/file.js

```diff
@@ -102,34 +123,40 @@ FileCache.prototype.getMetadata = function getMetadata (key) {
 FileCache.prototype.set = function set (key, data, options) {
   debug('SET %s %o', key, options)
 
-  return new Promise((resolve, reject) => {
-    let cachePath = this.getCachePath(key, options)
+  return this.db.then(db => {
+    db.findAndRemove({ $key: key })
 
-    // open a stream for writing the data
+    let cachePath = this.getCachePath(key, options)
     let cacheFile = fs.createWriteStream(cachePath, { flags: 'w' })
-
-    cacheFile.on('finish', () => {
-      return resolve('')
-    }).on('error', (err) => {
-      return reject(err)
-    })
-
     let stream
 
-    // create a stream from the data if it is a String or Buffer
+    // Create a stream from the data if it is a String or Buffer.
     if (data instanceof Buffer || typeof data === 'string') {
       stream = streamifier.createReadStream(data)
     } else if (data instanceof Stream) {
       stream = data
     }
 
-    stream.pipe(cacheFile)
-  }).then(result => {
-    if (!options.metadata) {
-      return result
-    }
+    // Adding to database.
+    let lastModified = Date.now()
+
+    db.insert({
+      $key: key,
+      $dadiCache: options.metadata || null,
```

Comment:
@eduardoboucas when attempting to get the tests to work, I had to add this line. Unsure if this is what was intended, but I'm sure you'll know what to do...

Approved 🌮